site-service/src/test/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceTests.java

```diff
@@ -0,0 +1,126 @@
+package uk.ac.ox.ndph.mts.site_service.service;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import uk.ac.ox.ndph.mts.site_service.exception.InitialisationError;
+import uk.ac.ox.ndph.mts.site_service.exception.ValidationException;
+import uk.ac.ox.ndph.mts.site_service.model.Site;
+import uk.ac.ox.ndph.mts.site_service.model.ValidationResponse;
+import uk.ac.ox.ndph.mts.site_service.repository.EntityStore;
+import uk.ac.ox.ndph.mts.site_service.validation.ModelEntityValidation;
+
+@ExtendWith(MockitoExtension.class)
+class SiteServiceTests {
+    
+    @Mock
+    private EntityStore<Site> siteStore;
+
+    @Mock
+    private ModelEntityValidation<Site> siteValidation;
+    
+    @Captor
+    ArgumentCaptor<Site> siteCaptor;
+
+    @Test
+    void TestSaveSiteParent_WithSiteParent_ValidatesSiteParent() {
+        // Arrange
+        String name = "name";
+        String alias = "alias";
+        String parent = "parent";
+
+        Site siteWithParent = new Site(name, alias, parent);
+        var siteService = new SiteService(siteStore, siteValidation);
+        when(siteValidation.validate(any(Site.class))).thenReturn(new ValidationResponse(true, ""));
+        when(siteStore.saveEntity(any(Site.class))).thenReturn("123");
+        //Act
+        siteService.saveSite(siteWithParent);
+
+        //Assert
+        Mockito.verify(siteValidation).validate(siteCaptor.capture());
+        var value = siteCaptor.getValue();
+        assertThat(siteWithParent, equalTo(value));
+    }
+
+    @Test
+    void TestSaveSite_WithSite_ValidatesSite() {
```

Comment:
Since the validation is a mock here, what value do you get from this test after you already tested the the validation gets called and receives the object?

Reply:
removed as this Test is not required